tests/app/dao/notification_dao/test_notification_dao.py

```diff
@@ -1670,3 +1671,13 @@ def test_send_method_stats_by_service(sample_service, sample_organisation):
         )
         == []
     )
+
+
+def test_bulk_insert_notification(sample_template):
+    assert len(Notification.query.all()) == 0
+    n1 = create_notification(sample_template, client_reference="happy", save_notification=False)
+    n2 = create_notification(sample_template, client_reference="sad", save_notification=False)
+    n3 = create_notification(sample_template, client_reference="loud", save_notification=False)
+    bulk_insert_notifications([n1, n2, n3])
+    all_notifications = get_notifications_for_service(sample_template.service_id).items
+    assert len(all_notifications) == 3
```

Comment:
What happens if a batch insert call is made with zero notifications? or if one faulty notification is inserted (but other two are fine)?

Reply:
What is an example of a faulty notification?

Reply:
Can you check how the save work and if you can trigger an error? If some field is missing for example? Then we can test if the whole batch fail or not.

Reply:
based on our offline convo, this will raise.  Added a todo here to update accordingly when we have redis functionality.